### DISCUSSION TITLE
E2E テスト内で fetcher によるモックデータ取得のエラーが出る問題について

### DISCUSSION THREAD

**Discussion by tuboihirokidesu** [tube-app/frontend-web#17] [Q&A]
## 作業ブランチ

[feat/e2e_in_comments](https://github.com/tube-app/frontend-web/tree/feat/e2e_in_commnets)

## 対応したこと

コメント画面において、ボタン押下して適切に画面遷移できているかなどのE2Eテストを書きました。

## 詰まっていること

`e2e > comments.spec.ts`のテストは現在通っているのですが、[comments.tsx](https://github.com/tube-app/frontend-web/blob/feat/e2e_in_commnets/app/comments/_components/comments.tsx)のデータをfetcher（fetch API の拡張）によるモックデータを呼び出すものからダミーデータベタ書きの状態に変えています。
fetcherのままでテストを実行すると

```sh
[WebServer]  ⨯ Internal error: Error: Proxy request aborted [GET http://localhost:3000/api/mock/comments?tab=core-fan]
    at handleFetch (/Users/tsuboihiroki/Desktop/next-js/tube-app-front-web/node_modules/.pnpm/next@14.0.4_@babel+core@7.23.3_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/experimental/testmode/fetch.js:111:19)
```
とエラーが出てその後タイムアウトでテストに失敗することになります。

fetcherのままでテストを通したいのですが、解決策が分かっていない状態です。
この問題に対するご助言やご提案をいただけますと大変ありがたいです。

**Comment by lightsound** [comment]
これ私の環境でそもそもテストが通らない（そのエラー以前にcreateEnvなどでこける）状況ですね。。。
修正しようにもなかなか難しそうなので、既存のテストが動くブランチを取り入れていただくことは可能ですか？

例えば私が出した下記などだとありがたいです🙏
https://github.com/tube-app/frontend-web/pull/20

**Reply by tuboihirokidesu** [reply]
@lightsound 
#20 取り入れました！
